Tests/Common/Securities/CashAccountMarginModelTests.cs

```diff
@@ -0,0 +1,326 @@
+            _spy = new QuantConnect.Securities.Equity.Equity(
+                SecurityExchangeHours.AlwaysOpen(tz),
+                new SubscriptionDataConfig(typeof(TradeBar), Symbols.SPY, Resolution.Minute, tz, tz, true, false, false),
+                _portfolio.CashBook[CashBook.AccountCurrency],
+                SymbolProperties.GetDefault(CashBook.AccountCurrency));
+
+            _btcusd = new Crypto(
+                SecurityExchangeHours.AlwaysOpen(tz),
+                _portfolio.CashBook[CashBook.AccountCurrency],
+                new SubscriptionDataConfig(typeof(TradeBar), Symbols.BTCUSD, Resolution.Minute, tz, tz, true, false, false),
+                SymbolProperties.GetDefault(CashBook.AccountCurrency));
+
+            _ethusd = new Crypto(
+                SecurityExchangeHours.AlwaysOpen(tz),
+                _portfolio.CashBook[CashBook.AccountCurrency],
+                new SubscriptionDataConfig(typeof(TradeBar), Symbols.ETHUSD, Resolution.Minute, tz, tz, true, false, false),
+                SymbolProperties.GetDefault(CashBook.AccountCurrency));
+
+            _btceur = new Crypto(
+                SecurityExchangeHours.AlwaysOpen(tz),
+                _portfolio.CashBook["EUR"],
+                new SubscriptionDataConfig(typeof(TradeBar), Symbols.BTCEUR, Resolution.Minute, tz, tz, true, false, false),
+                SymbolProperties.GetDefault("EUR"));
+
+            _ethbtc = new Crypto(
+                SecurityExchangeHours.AlwaysOpen(tz),
+                _portfolio.CashBook["BTC"],
+                new SubscriptionDataConfig(typeof(TradeBar), Symbols.ETHBTC, Resolution.Minute, tz, tz, true, false, false),
+                SymbolProperties.GetDefault("BTC"));
+
+            _marginModel = new CashAccountMarginModel(_portfolio.CashBook);
+        }
+
+        [TearDown]
+        public void TearDown()
+        {
+            _transactionHandler.Exit();
+        }
+
+        [Test]
+        public void InitializesCorrectly()
+        {
+            Assert.AreEqual(1m, _marginModel.GetLeverage(_spy));
+            Assert.AreEqual(1m, _marginModel.GetInitialMarginRequirement(_spy));
+            Assert.AreEqual(0m, _marginModel.GetMaintenanceMargin(_spy));
+            Assert.AreEqual(0m, _marginModel.GetMaintenanceMarginRequirement(_spy));
+        }
+
+        [Test]
+        public void SetLeverageDoesNotUpdateLeverage()
+        {
+            // leverage cannot be set, will always be 1x
+            _marginModel.SetLeverage(_spy, 50m);
+            Assert.AreEqual(1m, _marginModel.GetLeverage(_spy));
+        }
+
+        [Test]
+        public void EquityBuyOrderRequiresFullMargin()
+        {
+            _spy.SetMarketPrice(new Tick { Value = 250m });
+
+            var order = new MarketOrder(_spy.Symbol, 2, DateTime.UtcNow);
+            var initialMargin = _marginModel.GetInitialMarginRequiredForOrder(_spy, order);
+
+            Assert.AreEqual(501m, initialMargin);
```

Comment:
Comments on how you got `501` are a must in these types of tests. The tests serve as documentation, which, by the way, I do like how readable these tests are. I'm normally against running the world but it does make these tests easy to follow. I think we should just add a comment:
```csharp
// (current price * quantity) + fees = 501
```